File: Leetcode/75.cpp
class Solution {
public:
    void sortColors(vector<int> &nums) {
        unordered_map<int, int> A;
        A[0] = 0, A[1] = 0, A[2] = 0;
        for (const auto &i : nums) ++A[i];
        size_t i = 0;
        while (A[0] != 0) {
            nums[i++] = 0;
            --A[0];
        }
        while (A[1] != 0) {
            nums[i++] = 1;
            --A[1];
        }
        while (A[2] != 0) {
            nums[i++] = 2;
            --A[2];
        }
    }
};

File: Leetcode/75/75.go
func sortColors(nums []int)  {
	a, b, c := 0, 0, 0
	for i := 0; i < len(nums); i++ {
		if nums[i] == 0 {
			a++
		} else if nums[i] == 1 {
			b++
		} else {
			c++
		}
	}
	index := 0
	for a > 0 {
		nums[index] = 0
		index++
		a--
	}
	for b > 0 {
		nums[index] = 1
		index++
		b--
	}
	for c > 0 {
		nums[index] = 2
		index++
		c--
	}
}